pandas/core/indexes/category.py

```diff
@@ -781,6 +782,12 @@ def _delegate_method(self, name: str, *args, **kwargs):
             return res
         return CategoricalIndex(res, name=self.name)
 
+    def _wrap_joined_index(
+        self, joined: np.ndarray, other: "CategoricalIndex"
+    ) -> "CategoricalIndex":
+        name = get_op_result_name(self, other)
+        return self._create_from_codes(joined, name=name)
```

Comment:
i think this constructor is analogous to passing i8 to the DTI/TDI/PI constructors.  i wonder if there is some EA attribute/constructor that this corresponds to

Reply:
Not sure I understand what you mean, but those three all inherit from `DatetimeTimedeltaMixin` which has its own `_wrap_joined_index` method

Reply:
dont worry about it, was just spitballing, tangential to this PR